src/component/handlers/composite-android/DraftEditorCompositionHandlerAndroid.js

```diff
@@ -0,0 +1,198 @@
+  if (!mutationObserver) {
+    mutationObserver = new window.MutationObserver(handleMutations);
+  }
+};
+
+/**
+ * Replace the current selection with the specified text string, with the
+ * inline style and entity key applied to the newly inserted text.
+ */
+function replaceText(
+    editorState: EditorState,
+    text: string,
+    compositionRange: SelectionState,
+    // inlineStyle: DraftInlineStyle,
+    // entityKey: ?string,
+  ): EditorState {
+  var contentState = DraftModifier.replaceText(
+      editorState.getCurrentContent(),
+      compositionRange,
+      text,
+    //   inlineStyle,
+    //   entityKey,
+    );
+  return EditorState.push(editorState, contentState, 'insert-characters');
+}
+
+const getCompositionRange = (editor, text) => {
+  // Ocassionally a newline will get composed.  In this case, we want to strip it since
+  // we won't be able to match it in Draft, and it will get rewritten anyways.
+  if (text.endsWith('\n')) {
+    text = text.slice(0, text.length - 1);
+  }
+
+  if (!text) {
+    // get Selection (Assuming editorState is correct…)
+    // Since we know editorState is often out of sync right now, derive from the DOM:
+    const editorNode = ReactDOM.findDOMNode(editor.refs.editorContainer);
+    const draftSelection = getDraftEditorSelection(editor._latestEditorState, editorNode).selectionState;
+    return draftSelection;
+  } else {
+    // get Selection for text
+    const editorNode = ReactDOM.findDOMNode(editor.refs.editorContainer);
+    const draftSelection = getDraftEditorSelection(editor._latestEditorState, editorNode).selectionState;
+
+    const compositionRange = findCoveringIndex(editor._latestEditorState.getCurrentContent(), draftSelection, text);
+
+    return compositionRange;
+  }
+};
+
+function findCoveringIndex(contentState, selection, textToFind) {
+  if (!selection.isCollapsed()) {
+    // Expected a collapsed selection, return early
+    return selection;
+  }
+
+  const block = contentState.getBlockForKey(selection.getStartKey());
+  const blockText = block.getText();
+  const selStartOffset = selection.getStartOffset();
+  const matchStartOffset = blockText.indexOf(
+    textToFind,
+    // The earliest textToFind could start is its length before the selection
+    selStartOffset - textToFind.length,
+  );
+
+  if (
+    // Ensure the match exists and contains the selection
+    matchStartOffset >= 0 &&
+    matchStartOffset <= selStartOffset &&
+    matchStartOffset + textToFind.length >= selStartOffset
+  ) {
+    return selection.merge({
+      anchorOffset: matchStartOffset,
+      focusOffset: matchStartOffset + textToFind.length,
+    });
+  }
+
+  console.warn(
+    `findCoveringIndex: couldn't find index of '${textToFind}' in '${blockText}' after offset ${selStartOffset}!`,
+  );
+  return selection;
+}
+
+var DraftEditorCompositionHandlerAndroid = {
+  /**
+   * A `compositionstart` event has fired while we're still in composition
```

Comment:
I don't think this is entirely accurate since the `editOnCompositionStart` will reemit any `compositionstart` events through this `onCompositionStart` too.